pkg/cloud/aws/actuators/machine/instances.go

```diff
@@ -262,3 +263,38 @@ func launchInstance(machine *clusterv1.Machine, machineProviderConfig *providerc
 
 	return runResult.Instances[0], nil
 }
+
+type instanceList []*ec2.Instance
+
+func (il instanceList) Len() int {
+	return len(il)
+}
+
+func (il instanceList) Swap(i, j int) {
+	il[i], il[j] = il[j], il[i]
+}
+
+func (il instanceList) Less(i, j int) bool {
+	if il[i].LaunchTime == nil && il[j].LaunchTime == nil {
+		// No idea what to do here, should not be possible, so keep the order
```

Comment:
Does this actually happen?

Reply:
Dunno, just kept the comment from previous implementation.